rudder_airflow_provider/hooks/rudderstack.py

```diff
@@ -45,10 +49,13 @@ def poll_for_status(self):
         '''
         status_endpoint = f"/v2/sources/{self.source_id}/status"
         access_token = self.get_access_token()
+        headers = {
+            'authorization': f"Bearer {access_token}",
+            'Content-Type': 'text/plain'
```

Comment:
Same comment about content-type here. Ideally we should have a Client class that makes the calls to rudder-api. This will allow us to write more tests more efficiently (by writing integration tests for the client and mocking it for the rest of the unit tests), and also avoid repeating the code that prepares the request for each call.

Reply:
updated. 
Hook itself is the client interface for Rudderstack API within our package. Adding a method to prepare requests should be fine I guess?

Reply:
Thanks for letting me know. Yes, that's perfect